scan/scan.go

```diff
@@ -237,11 +247,51 @@ func (s *sourcescanner) walkSource(conn redis.Conn, path string, f os.FileInfo,
 		return nil, nil
 	}
 
+	var dfData DummyFile
+	dummyFile := GetConfig().DummyFiles
+
 	d := new(filedata)
 	d.path = path[len(GetConfig().Repository):]
-	d.size = f.Size()
-	d.modTime = f.ModTime()
 
+	if dummyFile {
+		fi, _ := os.Stat(d.path)
+		if fi.Size() > 1048576 {
+			log.Errorf("File is bigger than 1MB, falling back to normal")
+			goto skip
+		}
+		raw, err := ioutil.ReadFile(path)
+		if err != nil {
+			log.Errorf(err.Error())
+			log.Errorf("Failed to read file, falling back to normal")
```

Comment:
`Errorf` uses a Printf-like syntax, therefore your can replace this by:
```go
log.Errorf("Failed to read file: %s", err.Error())
```
But again, do we want to display an error in case of an hybrid repository? What about `log.Debugf`?